kubric/core.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2020 The Kubric Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import uuid
+from collections import defaultdict
+
+import traitlets as tl
+import mathutils
+from traitlets import default, validate
+
+import kubric.traits as kt
+from kubric.color import Color
+
+
+class Asset(tl.HasTraits):
```

Comment:
What abstraction does Asset provide? Is it mostly to support keyframes?

Reply:
Ensures everything has a `uid`, is hashable, and supports destructor callbacks and keyframing. 
More generally I wanted to have one base class for all the Kubric objects. So this is the smallest common denominator between 3D objects, materials, lights and cameras.